core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -79,6 +81,9 @@
         Setting.floatSetting("cluster.routing.allocation.balance.threshold", 1.0f, 0.0f,
             Property.Dynamic, Property.NodeScope);
 
+    public static final Setting<Float> INDEX_SHARD_WEIGHT_MULTIPLIER_SETTING =
```

Comment:
I'm more inclined to use floats as we use floats already in the settings of the balancing universe (for example `cluster.routing.allocation.balance.threshold`). I'm ok though with starting out with integers, which always give us the possibility to expand to floats in the future.